src/reducers/fileTree.tsx

```diff
@@ -466,18 +466,26 @@ export type FileTree = {
 };
 
 export type FileTreeState = {
+  comparedToVersionId: null | number;
   forVersionId: undefined | number;
   tree: undefined | FileTree;
 };
 
 export const initialState: FileTreeState = {
+  comparedToVersionId: null,
   forVersionId: undefined,
   tree: undefined,
 };
 
 export const actions = {
   buildTree: createAction('BUILD_TREE', (resolve) => {
-    return (payload: { version: Version }) => resolve(payload);
+    return (payload: {
+      comparedToVersionId: number | null;
+      version: Version;
+    }) => resolve(payload);
+  }),
+  unsetForVersionId: createAction('UNSET_FOR_VERSION_ID', (resolve) => {
```

Comment:
For the reasons I stated above, I think it's not a good idea to expose this action. Otherwise, it puts the burden of knowing when to unset the tree on each caller. The burden should be on this reducer (internally).